cmd/startup/startup.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	azstorage "github.com/Azure/azure-sdk-for-go/storage"
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/api/validate"
+	"github.com/openshift/openshift-azure/pkg/arm"
+	"github.com/openshift/openshift-azure/pkg/cluster"
+	azureclientstorage "github.com/openshift/openshift-azure/pkg/util/azureclient/storage"
+	"github.com/openshift/openshift-azure/pkg/util/cloudprovider"
+	"github.com/openshift/openshift-azure/pkg/util/configblob"
+	"github.com/openshift/openshift-azure/pkg/util/log"
+	"github.com/openshift/openshift-azure/pkg/util/template"
+)
+
+var (
+	logLevel  = flag.String("loglevel", "Debug", "valid values are Debug, Info, Warning, Error")
+	gitCommit = "unknown"
+)
+
+type startup struct {
+	blob azureclientstorage.Blob
+	log  *logrus.Entry
+}
+
+func (s *startup) writeTemplatedFiles(cs *api.OpenShiftManagedCluster) error {
+	tc := api.TestConfig{RunningUnderTest: cs.Config.RunningUnderTest}
+	hostname, _ := os.Hostname()
+	cname, _ := net.LookupCNAME(hostname)
+	domainname := strings.SplitN(strings.TrimSuffix(cname, "."), ".", 2)[1]
+
+	s.log.Infof("hostname is \"%s\"", hostname)
+	s.log.Infof("domainname %s", domainname)
+
+	files := []string{
```

Comment:
use arm.AssetNames() here, then blacklist etc/origin/node/pods/sync.yaml as appropriate (and I guess {master,node}-startup.sh)

Reply:
done